Commiting Intelledger changes

manthinaa opened the pull request:
Aaron, fixed your review comments and updated the pull request with new commit.

manthinaa committed:
Commiting Intelledger changes

aaron-brown commented on grails-app/domain/me/sudofu/intelledger/domain/IntelledgerEntry.groovy:
```diff
@@ -0,0 +1,86 @@
+    /**
+     * This describes whether the key is active or not.
+     */
+    boolean active
```
Change to `Boolean`; there are reasons to use the class over the simple datatype.

aaron-brown commented on grails-app/domain/me/sudofu/intelledger/domain/IntelledgerEntry.groovy:
```diff
@@ -0,0 +1,86 @@
+    /**
+     * This describes the version of the key.
+     */
+    int keyVersion
```
Change to `Integer version`, and see comments in the `mappings` below.

aaron-brown commented on grails-app/domain/me/sudofu/intelledger/domain/IntelledgerEntry.groovy:
```diff
@@ -0,0 +1,86 @@
+    static mapping = {
+        key         column: 'identifier'
+        value       type: "text"
+        description type: "text"
```
When you change the `keyVersion` to `version`, add the line `version column: 'entry_version'` below this line.

aaron-brown replied:
:+1: Done lower down.

aaron-brown commented on grails-app/domain/me/sudofu/intelledger/domain/IntelledgerEntry.groovy:
```diff
@@ -0,0 +1,86 @@
+
+    int compareTo(IntelledgerEntry obj) {
+        if(key.compareTo(obj.key) == 0) {
+            return keyVersion.compareTo(obj.keyVersion)
```
:+1: But remember to change `keyVersion` to `version`; this will make code look better.

aaron-brown commented on grails-app/domain/me/sudofu/intelledger/domain/Intelledger.groovy:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    static mapping = { description type: "text" }
+
```
Add the following method:

``` groovy
public List<IntelledgerEntry> getActiveEntries() {
    return entries?.findAll { it.active }
}
```

aaron-brown replied:
:+1: Done lower down.

aaron-brown commented on src/groovy/me/sudofu/intelledger/Intelledger.groovy:
```diff
@@ -0,0 +1,790 @@
+            return 0
+        }
+
+        return domain.entries.size()
```
After adding the `getActiveEntries()` function in **Intelledger**, replace this line with:

``` groovy
return domain.activeEntries.size()
```

aaron-brown commented on src/groovy/me/sudofu/intelledger/Intelledger.groovy:
```diff
@@ -0,0 +1,790 @@
+            return -1
+        }
+
+        if (domain.entries == null) {
```
After adding the `getActiveEntries()` function in **Intelledger**, replace this line with:

``` groovy
if (domain.activeEntries == null) {
```

manthinaa committed:
Fixed Aaron review comments -1

aaron-brown commented on src/groovy/me/sudofu/intelledger/Intelledger.groovy:
```diff
@@ -0,0 +1,790 @@
+     * exists, and it is associated with a <code>IntelledgerEntry</code>
+     * in which the <code>value</code> field matches that which was passed.
+     */
+    static boolean containsActive(String intelledgerName, value) {
```
Rename this to just `contains`, and change `domain.entries` to `domain.activeEntries`.

I want the perspective of these methods to be that the normal ones (`get`, `set`, `delete`) work on the assumed active entries, and thus other methods will be made to work on **all** entries.

aaron-brown commented on src/groovy/me/sudofu/intelledger/Intelledger.groovy:
```diff
@@ -0,0 +1,790 @@
+     * exists, and it is associated with a <code>SmartionaryEntry</code>
+     * in which the <code>key</code> field matches that which was passed.
+     */
+    static boolean containsActiveKey(String intelledgerName, String key) {
```
Rename to just `containsKey` and change `domain.entries` to `domain.activeEntries`.

aaron-brown commented on src/groovy/me/sudofu/intelledger/Intelledger.groovy:
```diff
@@ -0,0 +1,790 @@
+     * so it may be necessary to convert these values to other built-in
+     * datatypes (i.e. <code>Integer</code>) as desired / necessary.
+     */
+    static Map getActiveEntries(String intelledgerName) {
```
Rename to just `get`.

Commit:
Modifications and Implementations.

Intelledger (grails-app/domain):

* Added some additional getters, some of which are currently not used.

IntelledgerEntry:

* dos2unix (^M line-endings)
* Added another getter for readability.
* Altered the constraints so that `changeId` and `version` are **required**
fields, as to have that be optional defeats the purpose of Intelledger
(auditing).

Intelledger (src/groovy):

* Changed functions to move to a more "active-is-assumed" perspective;
that is, it is assumed in get, set, delete, etc. that these operations
are working with only the _active_ entries. No or few methods to
interact with all entries are present at this time.

* Method signatures reflect the required `changeId` field.

* Altered the `_set()` logic a little to shrink method size.

* Removed the ability to `delete()` an **Intelledger** itself, as there
is no way of auditing such an action and therefore defeats the purpose
of **Intelledger** to do so.

* Altered the behavior of the `delete()` functions, such that rather
than actually deleting the last entry, a new one is created with the
required `changeId` of the person who deleted it, with the same
information as the previous entry; both are marked _inactive_,
effectively rendering them deleted from all usual methods.

* JavaDoc updates (needs more updating, but this is at least a start).

manthinaa committed:
Commiting Aaron review comments -2

manthinaa committed:
Merge pull request #1 from aaron-brown/IntelledgerChanges-delta

Modifications and Implementations.
